### PR TITLE
Bug 1184166 - Add --quiet option

### PR DESCRIPTION
**Pull request by stasm**
https://bugzilla.mozilla.org/show_bug.cgi?id=1184166

### CODE REVIEW

**Comment by eliperelman** [comment]
Can you make a couple changes?

In the `bin/raptor` file, add an environment entry for `RAPTOR_QUIET`?

``` js
var environment = {
  ...
  RAPTOR_QUIET: ['quiet', false]
};
```

Then in `phase.js`, you should be able to remove the `.parent` qualifer:

``` js
if (this.options.parent.quiet) {

// goes to:

if (this.options.quiet) {
```

Then lastly would just update the help string to:

``` js
.option('-q, --quiet, RAPTOR_QUIET=<isQuiet>', 'suppress all output except for the results');
```

Can you try that?

**Comment by stasm** [comment]
Thanks for the pointers!  I'm getting the following error:

```
error: option `-q, --quiet, RAPTOR_QUIET=<isQuiet>' argument missing
```

I guess that's because of the `=` in the option definition.  We could try to find a way to support boolean options, _or_ change `quiet` to `output` which takes `normal` and `quiet` as value?  (In the future, you could add `debug` or `verbose` outputs, too.)

**Comment by eliperelman** [comment]
Our option parsing is totally broken right now, so let's hold off here until I fix that.

**Comment by stasm** [comment]
I rebased the PR on top of the current master.

File: bin/raptor [hunk 1]
```diff
@@ -82,6 +82,12 @@ var parent = parser
     help: 'path to store historical test metrics. Environment: RAPTOR_LOG',
     default: path.join(process.cwd(), 'raptor.log'),
     transform: utils.compose(utils.toAbsolute, utils.fromEnvironment('RAPTOR_LOG'))
+  })
+  .option('output', {
+    metavar: '<mode>',
+    help: 'output mode: normal or quiet. Environment: RAPTOR_OUTPUT',
+    default: 'normal',
```
Comment (eliperelman) [inline bin/raptor:89]: I think this is fine for now, and in the future if we add more choices we can switch to `nomnom` choices.

Comment (stasm) [inline bin/raptor:89]: Actually, I think it should be easy now to go back to the idea of the boolean --quiet option.  Which solution do you prefer?

Comment (eliperelman) [inline bin/raptor:89]: I prefer this, especially since this will enable more options later. :)

**Pull request merged**